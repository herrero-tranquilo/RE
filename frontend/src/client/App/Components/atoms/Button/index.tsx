import styled from "styled-components";

export default styled.button`
  outline: 0;
  border: 0;
  color: ${(props) => props.theme.blackColor};
  background-color: #fff;
  padding: 0px 7px;
  margin: 5px 5px;
  font-size: 14px;
  font-weight: 600;
  text-align: center;
  cursor: pointer;
`;
