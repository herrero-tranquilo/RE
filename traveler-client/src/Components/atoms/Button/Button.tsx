import React from "react";
import styled from "styled-components";
interface ISimpleProps extends React.HTMLAttributes<HTMLButtonElement> {
  className?: any;
  children: any;
  type?: "button" | "submit" | "reset";
  custom?: "default" | "wide";
}
const SimpleBtn = styled.button`
  outline: 0;
  border: 0;
  color: ${props => props.theme.blackColor};
  background-color: #fff;
  padding: 0px 7px;
  font-size: 14px;
  font-weight: 600;
  text-align: center;
  cursor: pointer;
`;
const WideBtn = styled(SimpleBtn)`
  width: 100%;
  border: 0;
  border-radius: 6px;
  color: white;
  font-weight: 600;
  background-color: ${props => props.theme.blackColor};
  text-align: center;
  padding: 7px 0px;
  font-size: 14px;
  margin: 5px 0;
  cursor: pointer;
`;
const Button = ({
  className,
  children,
  type = "button",
  custom,
  ...others
}: ISimpleProps) => {
  switch (custom) {
    case "wide":
      return (
        <WideBtn className={className} type={type} {...others}>
          {children}
        </WideBtn>
      );
    default:
      return (
        <SimpleBtn className={className} type={type} {...others}>
          {children}
        </SimpleBtn>
      );
  }
};

export default Button;
