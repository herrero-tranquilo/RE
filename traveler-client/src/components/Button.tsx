import React from "react";
import styled from "styled-components";

interface ISimpleProps {
  text: string;
  onClick?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
}
interface INormalProps extends ISimpleProps {
  type?: "button" | "submit" | "reset" | undefined;
}

const SimpleBtn = styled.span`
  color: ${props => props.theme.blackColor};
  padding: 0px 7px;
  font-size: 14px;
  font-weight: 600;
  text-align: center;
  cursor: pointer;
`;
const SimpleButton = ({ text, onClick }: ISimpleProps) => (
  <SimpleBtn onClick={onClick}>{text}</SimpleBtn>
);

const NormalBtn = styled.button`
  width: 100%;
  border: 0;
  border-radius: 6px;
  color: white;
  font-weight: 600;
  background-color: ${props => props.theme.blackColor};
  text-align: center;
  padding: 7px 0px;
  font-size: 14px;
  cursor: pointer;
`;

const NormalButton = ({ text, onClick, type }: INormalProps) => (
  <NormalBtn onClick={onClick} type={type}>
    {text}
  </NormalBtn>
);

export { NormalButton, SimpleButton };
