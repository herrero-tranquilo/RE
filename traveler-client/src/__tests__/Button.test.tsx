import React from "react";
import { mount, shallow, render } from "enzyme";
import Button from "../components/Button";

const clickFn = jest.fn();

describe("<Button />", () => {
  let button: any = null;

  it("renders correctly", () => {
    button = mount(<Button />);
  });

  it("matches snapshot", () => {
    expect(button).toMatchSnapshot();
  });

  it("span contain text 'btn'", () => {
    const btn = button.find("span");
    expect(btn.contains("btn")).toBe(true);
  });
});
