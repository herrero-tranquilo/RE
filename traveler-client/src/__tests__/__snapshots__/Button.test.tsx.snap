// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<Button>
  <span>
    btn
  </span>
</Button>
`;
